examples/p5LSystem.html

```diff
@@ -0,0 +1,175 @@
+	<!DOCTYPE html>
+<html>
+<head>
+  <meta charset="utf-8">
+	<title>SONIFYING LSYSTEMS</title>
+	<meta name="viewport" content="width=device-width, initial-scale=1, maximum-scale=1, user-scalable=no">
+
+	<script type="text/javascript" src="./scripts/p5.js"></script>
+	<script type="text/javascript" src="http://cdn.tonejs.org/r6/p5.Tone.js"></script>
+	<script type="text/javascript" src="./scripts/jquery.min.js"></script>
+	<!--<script type="text/javascript" src="./scripts/Interface.js"></script>
+
+	<link rel="stylesheet" type="text/css" href="./style/examples.css">-->
+
+	<script type="text/javascript">
+		// jshint ignore: start
+	</script>
+
+</head>
+<body>	
+	<script id="p5" type="text/javascript">
+	//based on the work of R. Luke Dubois
+	//http://www.music.columbia.edu/~luke/dissertation/dissertation.pdf
+	var generations = 5;
+
+	var axiom = "X";
+
+	var rules = {"X" : "F-[[X]+cX]+F[+FcX]-X", "F" : "FF"};
+	var instring, outstring;
+
+	var needsF = false;
+	var pitch = 60;
+	var position = 0;
+
+	var symMap = {
+		"F" : "C3",
+		"]" : "D3",
+		"+" : "F3",
+		"-" : "G3",
+		"c" : "A3",
+		"[" : "A#3"
+	}
+
+	function setup(){
+		createCanvas(windowWidth, windowHeight);
+
+		instring = axiom;
+		console.log("axiom: " + axiom);
+
+		//looping through the generations
+		for(var i = 0; i < generations; i ++){
+			outstring = "";
+			//looping through the input string
+			for(var j = 0; j < instring.length; j ++){
+				ismatching = false;
+				var currentChar = instring[j];
+				//checking to see if a rule applies
+				if(rules[currentChar]){
+					//if so, add the rule to the outstring
+					outstring = outstring + rules[currentChar];
+				} else {
+					//otherwise, just add current character to outstring
+					outstring = outstring + currentChar;
+				}
+			}
+			//set up our instring for the next iteration
+			instring = outstring;
+			console.log(i + ": " + outstring);
+		}
+
+		translate(width / 3, 2 * height / 3);
+		rotate(-45);
+
+	}
+
+	function draw(){
+		for(var i = 0, iMax = position; i < iMax; i ++){
+			turtle(instring[i]);
+		}
+		for(var i = 0; i < pushCount; i ++){
+			pop();
+		}
+		rotate(-totRotate);
+		totRotate = 0;
+		pushCount = 0;
+	}
+	</script>
+	<script id="Song" type="text/javascript">
+
+	synth = new Tone.FMSynth({
+		harmonicity: 8,
+		modulationIndex: 10,
+		carrier: {
+			oscillator: {
+				type: "sine"
+			},
+			envelope: {
+				release: 0.5
+			}
+		},
+		modulator: {
+			oscillator: {
+				type: "sine"
+			}
+		}
+	}).toMaster();
+
+	Tone.Transport.scheduleRepeat(function(time){
```

Comment:
This might be a good place to use Tone.Pattern instead and turn `instring` into an array of characters.